src/jsrs_parser.cc

```diff
@@ -418,51 +416,51 @@ static unsigned int ReadHexNumber(const char* str, size_t len, bool* ok);
 
 // Parses a part of a JavaScript string representation after the backslash
 // character (i.e., an escape sequence without \) into an unescaped control
-// character.
-static char* GetControlChar(Isolate*    isolate,
-                            const char* str,
-                            size_t*     res_len,
-                            size_t*     size) {
-  char* result = new char[5];
+// character and writes it to `write_to`.
+// Returns true if no error occured, false otherwise.
+static bool GetControlChar(Isolate*    isolate,
+                           const char* str,
+                           size_t*     res_len,
+                           size_t*     size,
+                           char* write_to) {
   *size = 1;
   *res_len = 1;
   bool ok;
   switch (str[0]) {
     case 'b': {
-      *result = '\b';
+      *write_to = '\b';
       break;
     }
     case 'f': {
-      *result = '\f';
+      *write_to = '\f';
       break;
     }
     case 'n': {
-      *result = '\n';
+      *write_to = '\n';
       break;
     }
     case 'r': {
-      *result = '\r';
+      *write_to = '\r';
       break;
     }
     case 't': {
-      *result = '\t';
+      *write_to = '\t';
       break;
     }
     case 'v': {
-      *result = '\v';
+      *write_to = '\v';
       break;
     }
     case '0': {
```

Comment:
*(Not related to this PR)*
Do we really need `'\0'` as a special case here? Either we support any octal sequences, or we don't support them at all.